### PR TITLE
[128] - Docs for Voice Service

### PR DESCRIPTION
**Pull request by MustafaAkolawala**
Reviewer: @amiraliemami 
Estimate: 40 mins

---

## Ticket

Fixes: [JIRA_TICKET_LINK](https://idinsight.atlassian.net/browse/AAQ-801?atlOrigin=eyJpIjoiMWY1ZTJjMjhlY2IyNDI5OGE2NzRmZGEyZTY2ODllNTYiLCJwIjoiaiJ9)

## Description
This pull request adds comprehensive documentation for the speech functionality in AAQ . The documentation covers three main areas:

1. `/voice-search` endpoint in Question Answer Service
2. External and internal speech models in optional components
3. Instructions for performing speech tests

### Goal
The goal of this pull request is to add comprehensive documentation for the speech functionality in AAQ. This documentation aims to provide clear guidance on the `/voice-search` endpoint, external and internal speech models, and the process of performing speech tests.

### Changes
- Added documentation for the `/voice-search` endpoint in the Question Answer Service
- Documented external and internal speech models in the optional components
- Added instructions for performing speech tests

### Future Tasks (optional)
Add documentation for manual setup for in-house models when manual setup is ready

## How has this been tested?
Manually reviewed all added documentation for accuracy and completeness

## Checklist

Fill with `x` for completed. 

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts
- [x] I have updated affected documentation

### CODE REVIEW

File: docs/develop/testing/speech-service-testing.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+## Running unit tests
+
+**Environment Setup**
+
+Please use the `aaq-speech` environment to run speech tests. You can create this environment using the following command from the root directory:
+
+```shell
+make fresh-env-speech
+```
```
Comment (amiraliemami) [inline docs/develop/testing/speech-service-testing.md:18]: this doesnt exist in the repo?

File: docs/develop/testing/speech-service-testing.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+## Executing Tests
+
+To run the tests, use the following command:
+
+```shell
+make tests
+```
```
Comment (amiraliemami) [inline docs/develop/testing/speech-service-testing.md]: Let's first mention to "navigate to the speech service with `cd optional_components/speech_api`" and put the `make fresh-env-speech` inside the makefile there only?

**Review by amiraliemami — APPROVED** [review]
Thanks for this! Left a few comments and made some changes since they were easy to make. Approved - pls merged after having a look at the comments :)

File: docs/components/voice-service/in-house-models.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# How to use in-house Speech models
+
+This guide outlines the process for hosting and utilizing our custom In-house **Speech-to-Text** and **Text-to-Speech** models using a specialized Docker image designed for end-to-end Speech Service.
+
+## Prerequisite steps
+
+### **Step 1:** Configure `Environment Variables`.
+
+To access the in-house models, ensure that the `CUSTOM_TTS_ENDPOINT` and `CUSTOM_STT_ENDPOINT` environment variables are properly set. These should be configured in the `.core_backend.env` file (cf. [Configuring AAQ](../../deployment/config-options.md#configuring-the-backend-core_backend)).
```
[the remark this answers is not in the document]
Comment (MustafaAkolawala) [reply docs/components/voice-service/in-house-models.md:9]: hmm , for the inhouse models all they really need to do is set these 2 environment variables which are mentioned, are u suggesting to tell them about different usecases like using inhouse STT model and then using external TTS model and vice versa like that?

File: docs/components/qa-service/voice-search.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    AAQ->>Vector DB: Request N most similar contents in DB
+    Vector DB->>AAQ: <N contents with similarity score>
+    AAQ->>Cross-encoder: Re-rank to get top N contents
+    Cross-encoder->>AAQ: <N contents with similarity score>
+    AAQ->>LLM: Generate response based on top N contents
+    LLM->>AAQ: <Text response>
+    AAQ->>Text-to-Speech: Convert text response to audio
+    Text-to-Speech->>AAQ: Audio file URL
+    AAQ->>User: Return JSON with text response, audio URL, and N contents
+```
```
[the remark this answers is not in the document]
Comment (MustafaAkolawala) [reply docs/components/qa-service/voice-search.md:30]: yes sure!

[the remark this answers is not in the document]
Comment (amiraliemami) [reply docs/components/qa-service/voice-search.md:30]: did you manage to add this? I mean like a vertical line item to show where we store the intermediary files

File: docs/components/voice-service/in-house-models.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (amiraliemami) [reply docs/components/voice-service/in-house-models.md:9]: No what I mean is to explain what "properly set" means. Like explain what the values should be and do

File: docs/develop/testing/speech-service-testing.md (hunk 1 above)
Comment (amiraliemami) [reply docs/develop/testing/speech-service-testing.md:18]: ohh

**Pull request merged**